src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -390,6 +393,37 @@ public RealJenkinsRule withHost(String host) {
         return this;
     }
 
+    /**
+     * Removes the prefix from the Jenkins root URL.
+     * @see #withPrefix(String)
+     */
+    public RealJenkinsRule noPrefix() {
+        this.prefix = null;
+        return this;
+    }
+
+    /**
+     * Sets a custom prefix for the Jenkins root URL.
+     * <p>
+     * By default, the prefix defaults to {@code /jenkins}.
+     * <p>
+     * Use {@link #noPrefix()} to remove the prefix, or provide another prefix of your choice as long as it starts with / and doesn't end with /.
+     */
+    public RealJenkinsRule withPrefix(@NonNull String prefix) {
+        var sanitized = Util.fixEmpty(prefix);
+        if (sanitized == null) {
+            throw new IllegalStateException("Use noPrefix() to remove the prefix.");
+        }
```

Comment:
To my mind it would be clearer to delete `noPrefix` and just let `.withPrefix("")` mean to request `http://localhost:12345/` which would be consistent with `.withPrefix("/xxx")` meaning `http://localhost:12345/xxx/`. The root URL is always expected to end with a slash.